plugins/inputs/intel_powerstat/sample.conf

```diff
@@ -25,3 +25,7 @@
   ##   "cpu_busy_frequency"
   ## ATTENTION: cpu_busy_cycles is DEPRECATED - use cpu_c0_state_residency
   # cpu_metrics = []
+
+  # The user can set the timeout duration in milliseconds for MSR reading
+  # The Default value is 100 ms.
+  # msr_read_timeout_ms = 100
```

Comment:
Shouldn't we have support for no timeout (or maybe even have it be the default value?) to be backward compatible?

We could still document that 100ms would be a sane value.

Reply:
@p-zak thoughts on a sane default?

Reply:
`100 ms` seems like a reasonable value. Additionally, it would be good if the code were able to handle a value of 0, which would disable this timeout. In this case, the plugin would behave the same as it has so far (simply reading the value and handling errors - without creating channels and additional goroutines).

So we will have two options to choose from:

We maintain backward compatibility. The default will be `0`, and the code will work as it has been. In the README (and sample.conf), it will be mentioned that in the case of increased load where the plugin hangs, the user can set the timeout to `100 ms`.
We break backward compatibility. The default will be `100 ms`.
I slightly lean towards option number 1.

@Hipska @powersj What do you think?

Reply:
agreed with option 1 - thanks!